plutus-tutorial/doc/plutus-tx.adoc

```diff
@@ -0,0 +1,383 @@
+the integer `1`.
+
+NOTE: We've included doctests to the examples that show the Plutus Core that is
+generated from compilation. The syntax of Plutus Core will look unfamiliar. This is fine, since it is
+the "assembly language" and you won’t need to inspect the output of
+the compiler. However, for our purposes it’s
+instructive to look at it to get a vague idea of what’s going on.
+
+[source,haskell]
+----
+integerOne :: CompiledCode Integer
+integerOne = $$(compile -- <.>
+    [|| (1 :: Integer) ||]) -- <.> <.>
+
+{- |
+>>> pretty $ getPlc integerOne
+(program 1.0.0
+  (con 1)
+)
+-}
+----
+<.> `compile` turns the `TExpQ Integer` into a `TExpQ (CompiledCode Integer)`,
+and the splice inserts the `TExpQ (CompiledCode Integer)` into the program.
+<.> The quote has type `TExpQ Integer`.
+<.> We always use unbounded integers in Plutus Core, so we have to pin
+down this numeric literal to an `Integer` rather than an `Int`.
+
+We can see how the metaprogramming works here: the Haskell program `1`
+was turned into a `CompiledCode Integer` at compile time, which we
+spliced into our Haskell program, and which we can then inspect at
+runtime to see the generated Plutus Core (or to put it on the
+blockchain).
+
+We also see the standard usage pattern here: a TH quote, wrapped in a call to `compile`,
+wrapped in a `$$` splice. This is how we write all of our Plutus Tx
+programs.
+
+Here’s a slightly more complex program, namely the identity function on
+integers.
+
+[source,haskell]
+----
+integerIdentity :: CompiledCode (Integer -> Integer)
+integerIdentity = $$(compile [|| \(x:: Integer) -> x ||])
+
+{- |
+>>> pretty $ getPlc integerIdentity
+(program 1.0.0
+  (lam ds (con integer) ds)
+)
+-}
+----
+
+== Functions and datatypes
+
+You can use functions inside your expression. In practice, you
+will usually want to define the entirety of your Plutus Tx program as a
+definition outside the quote, and then simply call it inside the quote.
+
+[source,haskell]
+----
+{-# INLINABLE plusOne #-} -- <.>
+plusOne :: Integer -> Integer
+plusOne x = x `addInteger` 1 -- <.>
+
+{-# INLINABLE myProgram #-}
+myProgram :: Integer
+myProgram =
+    let
+        plusOneLocal :: Integer -> Integer -- <.>
+        plusOneLocal x = x `addInteger` 1
+
+        localTwo = plusOneLocal 1
+        externalTwo = plusOne 1
+    in localTwo `addInteger` externalTwo
+
+functions :: CompiledCode Integer
+functions = $$(compile [|| myProgram ||])
+
+{- |
+>>> pretty $ runCk $ getPlc functions -- <.>
+(con 4)
+-}
+----
+<.> Functions which will be used in Plutus Tx programs should be
+marked with GHC’s `INLINABLE` pragma.
+This is usually necessary for non-local functions to
+be usable in Plutus Tx blocks, as it instructs GHC to keep the
+information that the Plutus Tx compiler needs. While you may be able
+to get away with omitting it, it is good practice to always include it.
+<.> Local functions do not need to be marked as `INLINABLE`.
```

Comment:
This comment and the next one are in the wrong order - it should be `addInteger` first, and then "Local functions ..."